partiql-parser/src/main/kotlin/org/partiql/parser/internal/PartiQLParserDefault.kt

```diff
@@ -1738,15 +1736,11 @@ internal class PartiQLParserDefault : PartiQLParser {
         }
 
         override fun visitCanCast(ctx: GeneratedParser.CanCastContext) = translate(ctx) {
-            val expr = visitExpr(ctx.expr())
-            val type = visitAs<Type>(ctx.type())
-            exprCanCast(expr, type)
+            throw error(ctx, "CAN_CAST is no longer supported in the default PartiQLParser")
         }
 
         override fun visitCanLosslessCast(ctx: GeneratedParser.CanLosslessCastContext) = translate(ctx) {
-            val expr = visitExpr(ctx.expr())
-            val type = visitAs<Type>(ctx.type())
-            exprCanLosslessCast(expr, type)
+            throw error(ctx, "CAN_LOSSLESS_CAST is no longer supported in the default PartiQLParser")
```

Comment:
(self-review): erroring on `canCast` and `canLosslessCast` in the partiql-parser ANTLR visitor does not result in any parsing test failures. As we alter the parsing behavior for other nodes we want to delete from `partiql-ast`, we may encounter some parsing test failures. We will need some skip-list or other mechanism to
- not run the PartiQLParserDefault on those tests
- continue to support parsing w/ the PartiQLPigVisitor